flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/SlotProfile.java

```diff
@@ -129,18 +116,47 @@ public static SlotProfile noLocality(ResourceProfile resourceProfile) {
 	 * @param preferredLocations specifying the preferred locations
 	 * @return Slot profile with the given resource profile and preferred locations
 	 */
-	public static SlotProfile preferredLocality(ResourceProfile resourceProfile, Collection<TaskManagerLocation> preferredLocations) {
-		return new SlotProfile(resourceProfile, preferredLocations, Collections.emptyList());
+	@VisibleForTesting
+	public static SlotProfile preferredLocality(
+			final ResourceProfile resourceProfile,
+			final Collection<TaskManagerLocation> preferredLocations) {
+
+		return priorAllocation(resourceProfile, preferredLocations, Collections.emptyList());
 	}
 
 	/**
 	 * Returns a slot profile for the given resource profile and the prior allocations.
 	 *
 	 * @param resourceProfile specifying the slot requirements
+	 * @param preferredLocations specifying the preferred locations
 	 * @param priorAllocations specifying the prior allocations
-	 * @return Slot profile with the given resource profile and prior allocations
+	 * @return Slot profile with the given resource profile, preferred locations and prior allocations
 	 */
-	public static SlotProfile priorAllocation(ResourceProfile resourceProfile, Collection<AllocationID> priorAllocations) {
-		return new SlotProfile(resourceProfile, Collections.emptyList(), priorAllocations);
+	@VisibleForTesting
+	public static SlotProfile priorAllocation(
```

Comment:
Ah yes! Didn't noticed that it can be package private.
It is used a few times in `PreviousAllocationSlotSelectionStrategyTest` for testing purpose. I think we can just remove it and refactor its usages to the other `priorAllocation(...)` method.

Reply:
done.